components/loki.libsonnet

```diff
@@ -1,28 +1,62 @@
-{
-  local loki = self,
+// These are the defaults for this components configuration.
+// When calling the function to generate the component's manifest,
+// you can pass an object structured like the default to overwrite default values.
+local defaults = {
+  local defaults = self,
+
+  name: error 'must provide name',
+  namespace: error 'must provide namespace',
+  version: error 'must provide version',
+  image: error 'must provide image',
+  replicas: error 'must provide replicas',
+  objectStorageConfig: error 'must provide object storage config',
+  queryConcurrency: 32,
+  queryParallelism: 32,  // Defaults to queryConcurrency because single query-frontend replica
+  ports: {
+    gossip: 7946,
+  },
```

Comment:
What is the reason to expose this port per config?

Reply:
I don't actually get the question, can you maybe rephrase? This is not exposed. This just a top-level configuration that we utilize whenever we need to access the ports.

Reply:
My intention was to say, that folks using the new pattern should be able to override this port.

Reply:
👍 Yes, it's totally possible and cleaner.